### ISSUE TITLE
tsh config should use web proxy address

### ISSUE THREAD

**Issue by programmerq** [gravitational/teleport#16179]
### Expected behavior:

When `tsh config` is run, it should use the web proxy addr for the `--proxy` argument in the ProxyCommand output.

### Current behavior:

For a cluster that has a different address for the web proxy addr and ssh proxy addr, `tsh config` writes out the ssh proxy addr for the `--proxy` argument. This is incorrect.

### Bug details:
- Teleport version - `Teleport v10.1.9 git:v10.1.9-0-g8f6f6ccbb go1.18.3`
- Recreation steps - stand up a cluster with different DNS names for web and ssh addresses. Run `tsh config` and observe that it refers to the ssh addr.
- Debug logs - n/a

This line of code appears to be the culprit:

https://github.com/gravitational/teleport/blob/45c977aea8d63846095ea4f50cc28bd2c8da179f/tool/tsh/config.go#L80

I believe it would be fixed by changing it to:

```go
	proxyHost, _, err := net.SplitHostPort(tc.Config.WebProxyAddr)
```

**Comment by stevenGravy** [comment]
@programmerq , `tsh config` doesn't output a port so I believe this can be closed.

**Comment by stevenGravy** [comment]
Example output:

```
# Common flags for all enterprise.teleport.sh hosts
Host *.example.teleport.sh example.teleport.sh
    UserKnownHostsFile "/Users/example/.tsh/known_hosts"
    IdentityFile "/Users/example/.tsh/keys/example.teleport.sh/example"
    CertificateFile "/Users/example/.tsh/keys/example.teleport.sh/example-ssh/example.teleport.sh-cert.pub"
    PubkeyAcceptedKeyTypes +ssh-rsa-cert-v01@openssh.com
    HostKeyAlgorithms ssh-rsa-cert-v01@openssh.com

# Flags for all enterprise.teleport.sh hosts except the proxy
Host *.example.teleport.sh !example.teleport.sh
    Port 3022
    ProxyCommand "/tmp/10/teleport/tsh" proxy ssh --cluster=example.teleport.sh --proxy=example.teleport.sh %r@%h:%p
```

**Comment by programmerq** [comment]
I believe this was prompted by a case where someone had a different public_addr and web_public_addr in their config, so it differed by more than just the port number. Past me should have included that detail!

**Comment by zmb3** [comment]
I think @stevenGravy fixed this in #27536

**Comment by stevenGravy** [comment]
Fixed in #27536
